crates/bevy_ecs/src/query/iter.rs

```diff
@@ -174,8 +174,8 @@ where
     /// # Safety
     /// This does not check for mutable query correctness. To be safe, make sure mutable queries
     /// have unique access to the components they query.
-    /// This does not validate that `world.id()` matches `query_state.world_id`. Calling this on a `world`
-    /// with a mismatched WorldId is unsound.
+    /// This does not validate that `world.id()` matches `query_state.world_id`. Calling this on a
+    /// `world` with a mismatched [`WorldId`] is unsound.
```

Comment:
```suggestion
    /// `world` with a mismatched [`WorldId`][crate::world::WorldId] is unsound.
```

(This is also a private item)